src/admin.rs

```diff
@@ -43,10 +46,10 @@ const RESOLUTION_KEY: &str = "resolution";
 const LSR_KEY: &str = "lsr";
 const REPLICAS_KEY: &str = "replicas";
 
-pub struct FluvioAdminWrapper(SharedFluvio);
+pub struct FluvioAdminWrapper(Arc<RwLock<FluvioAdmin>>);
```

Comment:
You shouldn't need to wrap in the lock.

Reply:
I tried without the lock, but Fluvio::admin uses `&mut self`

```rust
pub async fn admin(&mut self) -> FluvioAdmin {
        FluvioAdmin::new(self.create_serial_client().await)
    }
```

Reply:
Same with admin::list

```rust
pub async fn list<S, F>(&mut self, filters: F) -> Result<Vec<Metadata<S>>, FluvioError>
```

Reply:
Why do you  need to create admin here?, you already wrapped instance FluvioAdmin.   This method should not exist.

Reply:
commit 4c65c20 fixes;